src/09-word-problems.js

```diff
@@ -127,7 +159,30 @@ function compareLocations(address1, address2) {}
     //>   },
     //> ];
  */
-function gradeAssignments(assignments) {}
+function gradeAssignments(assignments) {
+  
+  for (let i = 0; i < assignments.length; i++){
+    if (assignments[i].kind === "PASS-FAIL") {
+      if (assignments[i].score.received === assignments[i].score.max) {
+        assignments[i].status = "PASSED";
+      } else {
+        assignments[i].status = "FAILED";
+      }
+    } else if (assignments[i].kind === "PERCENTAGE") {
+      if ((assignments[i].score.received / assignments[i].score.max) >= 0.8) {
+        assignments[i].status = `PASSED: ${((assignments[i].score.received / assignments[i].score.max) * 100).toFixed(1)}%`;
+      } else {
+        assignments[i].status = `FAILED: ${((assignments[i].score.received / assignments[i].score.max) * 100).toFixed(1)}%`;
+      }
+    } else {
+        assignments[i].status = `SCORE: ${assignments[i].score.received}/${assignments[i].score.max}`;
+    }  
+      
+  } 
+  
+      return assignments;
+
+}
```

Comment:
Nested `if/else` chains are easy to have a bug in and difficult to update. Stretch yourself by making the main `if/else-if` chain into a `switch` statement!